firmware-nucleo/smart-air-hockey-table/App/Inc/driver_goaldetection.h

```diff
@@ -0,0 +1,17 @@
+#ifndef INC_DRIVER_GOALDETECTION_H_
+#define INC_DRIVER_GOALDETECTION_H_
+
+#include <stdbool.h>
+#include <inttypes.h>
+
+typedef enum _ScoreState {
+	ScoreState_Clear,	// Neither player has scored
+	ScoreState_A,		// Player A has scored
+	ScoreState_B		// Player B has scored
+} ScoreState;

```

Comment:
Wouldn't we rather make the change directly to game state? i.e. calling
```
App_StateMachine_SetState(GameState_PlayerAScored)
```
[Look here](https://github.com/alanssitis/smart-air-hockey-table/pull/87#issuecomment-1765547660)

This way we can have all logic be handled by a single entity.

Reply:
I was worried about issues with multiple interrupts being raised, especially if we are trying to distinguish between the score and win states.

Reply:
you can go from scored -> change score -> change to appropriate state.
else, I think we can be very prone to race conditions.